### PR TITLE
Fix documentation

### PR DESCRIPTION
**Pull request by PerchunPak**
This PR fixes merge conflicts between #477 and #306.

I also moved `mcstatus.dns` out of Public API because of two possible situations:

1. We do a new release, where `mcstatus.dns` is a part of Public API. Then we continue a conversation, where everyone agrees that this shouldn't be a part of Public API. In the result, we remove `mcstatus.dns` from the Public API and do a breaking change.
2. We remove `mcstatus.dns` for now, but then everyone agrees that it should be a part of Public API. We do a new release, where is no breaking change, only minor version bump.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -103,10 +104,6 @@ def _get_project_meta() -> dict[str, str]:
 
 add_module_names = False
 
-autodoc_default_options = {
-    "show-inheritance": True,
-}
-
```
Comment (ItsDrike) [inline docs/conf.py:109]: Why was this removed? Do we not want to show inheritance anymore?

Comment (PerchunPak) [reply docs/conf.py:109]: I do not like to see `Bases: ...` on response classes, as this gives no info for the user (base classes are not a part of Public API).

Also, there is `:show-inheritance:` almost on every class.

**Review by ItsDrike — APPROVED** [review]

File: docs/api/basic.rst [hunk 2]
```diff
@@ -34,113 +34,31 @@ These are the classes that you get back after making a request.
...
-        :canonical: mcstatus.pinger.PingResponse.version
-
-        The version information.
-
-    .. attribute:: description
-        :type: str
-        :canonical: mcstatus.pinger.PingResponse.description
-
-        The description of the server. Also known as MOTD.
+.. autoclass:: mcstatus.status_response.JavaStatusResponse()
```
Comment (kevinkjt2000) [inline docs/api/basic.rst:39]: I love it!

**Review by kevinkjt2000 — APPROVED** [review]

File: docs/api/basic.rst (hunk 2 above)
Comment (ItsDrike) [reply docs/api/basic.rst:39]: Indeed, good stuff, thanks!
